server/src/domain/download/download.service.ts

```diff
@@ -107,6 +107,7 @@ export class DownloadService {
       const assetIds = dto.assetIds;
       await this.access.requirePermission(auth, Permission.ASSET_DOWNLOAD, assetIds);
       const assets = await this.assetRepository.getByIds(assetIds);
+      // eslint-disable-next-line @typescript-eslint/require-await
```

Comment:
you might be able to `yield Promise<assets>` and remove the async.

Reply:
Tried it, but 

 Property '[Symbol.asyncIterator]' is missing in type 'Generator<{ new (executor: (resolve: (value: AssetEntity[] | PromiseLike<AssetEntity[]>) => void, reject: (reason?: any) => void) => void): Promise<AssetEntity[]>; ... 7 more ...; readonly [Symbol.species]: PromiseConstructor; }, void, unknown>' but required in type 'AsyncGenerator<AssetEntity[], any, unknown>'.ts(2741)

Reply:
Ahh, it might need async to be an actual async generator, opposed to a normal generator that yields a promise.